### DISCUSSION TITLE
File Naming Conventions

### DISCUSSION THREAD

**Discussion by brownts** [WebFreak001/code-debug#351] [General]
Most of the files in the current code base are only single word files.  I'm planning to add a new multi-word file as part of a bug fix and trying to decide what the correct filename convention should be.  I looked around the code base for some guidance, but most files are single word only.  I did find a few examples, but they appear to be inconsistent:
- mi_parser.ts
- gdb_expansion.ts
- mi2mago.ts
- mi2lldb.ts

Is there any guidance or preference for file naming conventions?  Seems like there are at least the following options:
- PascalCase
- camelCase
- snake_case
- kebab-case
- alloneword

From the code base examples shown above, they seem to fall into the "snake_case" or "alloneword" scenarios.  Any thoughts on this?

**Comment by GitMensch** [comment]
I'm totally into using full words (seldom abbreviation) and when using multiple words separating them with an underscore.
[Reasoning](https://www.gnu.org/prep/standards/html_node/Names.html).
